scipy/stats/_continuous_distns.py

```diff
@@ -3247,6 +3247,114 @@ def _ppf(self, q, k, s):
 mielke = mielke_gen(a=0.0, name='mielke')
 
 
+class kappa4_gen(rv_continuous):
+    """Kappa 4 parameter distribution
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for kappa4 is::
+
+        kappa4.pdf(x, h, k) = (1.0 - k*x)**(1.0/k - 1.0)*
+                              (1.0 - h*(1.0 - k*x)**(1.0/k))**(1.0/h-1)
+
+    if ``h`` and ``k`` are not equal to 0.
+
+    kappa4 takes ``h`` and ``k`` as shape parameters, skew and
+    kurtosis respectively.
+
+    The kappa4 distribution return other distributions when certain
+    ``h`` and ``k`` values are used.
+
+    +------+-------------+--------------+---------------+
+    | h    | k=0.0       | k=1.0        | -2<=k<=2      |
+    +======+=============+==============+===============+
+    |  1.0 | Exponential | Uniform      | Generalized   |
+    |      |             |              | Pareto        |
+    +------+-------------+--------------+---------------+
+    |  0.0 | Gumbel      | Reverse      | Generalized   |
+    |      |             | Exponential  | Extreme Value |
+    |      |             | (note:not in |               |
+    |      |             | scipy)       |               |
+    +------+-------------+--------------+---------------+
+    | -1.0 | Logistic    |              | Generalized   |
+    |      |             |              | Logistic      |
+    +------+-------------+--------------+---------------+
+
+    %(after_notes)s
+
+    %(example)s
+
+    """
+    def _argcheck(self, h, k):
+        return (h >= -2) & (h <= 2) & (k >= -2) & (k <= 2)
```

Comment:
Indeed, there are upper and lower bounds dependent on h and k, but at least as far as I could figure out it would be difficult to implement in scipy since loc and scale are part of the transformed x before any of the "_*" methods are called.

ξ + α ( 1 – h –k) / k ≤ x ≤ ξ + α/k    if h > 0 & k > 0
ξ + α ln(h) ≤ x ≤ ∞                          if h > 0 & k = 0
ξ + α ( 1 – h –k) / k ≤ x ≤ ∞            if h > 0 & k < 0
– ∞ < x ≤ ξ + α/k                            if h ≤ 0 & k > 0
– ∞ < x < ∞                                    if h ≤ 0 & k = 0
ξ + α/k ≤ x < ∞                               if h ≤ 0 & k < 0

from: http://etd.lsu.edu/docs/available/etd-05182004-144851/unrestricted/Finney_dis.pdf